CONTRIBUTING.md

```diff
@@ -72,11 +190,71 @@ See [here](guidelines/python-cli-ux-guidelines.md) for full UX guidelines to fol
-ensuring they pass
+4. Run tests as described in the [testing section of this document](https://github.com/cyberark/conjur-api-python3/blob/master/CONTRIBUTING.md#testing), ensuring they pass
 5. Submit a pull request, linking the issue in the description
 6. Adjust labels as-needed on the issue. Ask another contributor to review and merge your code if there are delays in merging.
 
 ## Releasing
 
-TODO: Define release workflow once we have a PyPI publishing in place
+To create a tag and release follow the instructions in this section.
+
+### Update the version, changelog, and notices
+
+1. Create a new branch for the version bump
+
+1. Based on the unreleased content, determine the new version number and update the version in `version.py`
+
+1. Review the git log and ensure the changelog contains all relevant recent changes with references to GitHub issues or PRs, if possible
+
+1. Review the changes since the last tag, and if the dependencies have changed revise the [NOTICES](NOTICES.txt) file to correctly capture the added dependencies and their licenses / copyrights
+
+1. Before creating a release, ensure that all documentation that needs to be written has been written by TW, approved by PO/Engineer, and pushed to the forward-facing documentation
+
+1. Commit these changes to the branch. Bump version to x.y.z is an acceptable commit message and open a PR for review
+
+Add a git tag
+
+1. Once your changes have been reviewed and merged into master, tag the version using `git tag -s v0.1.1`. Note this requires you to be able to sign releases. Consult the [github documentation on signing commits](https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/managing-commit-signature-verification) on how to set this up vx.y.z is an acceptable tag message
+1. Push the tag: `git push vx.y.z` (or `git push origin vx.y.z` if you are working from your local machine)
+
+### Add release artifacts
+
+Currently, packing the client into an executable is a manual process. For Linux and Windows, you will need to pack the client using the different VMs we have available to us. For macOS, you will need to use your local machine.
+
+#### For all OS types:
+
+1. Clone the repo by running `git clone https://github.com/cyberark/conjur-api-python3.git`
+1. Install [Git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git) and [Python](https://realpython.com/installing-python/) if not already on the machine
+1. Activate [venv](#development) and install the requirements
+
+#### Linux
+
+1. Run `./bin/build_binary -l`. Once this is run, a `dist` folder will be created with the executable in it
+2. Once an executable has been created, run `tar -cvf conjur-cli-linux.tar conjur` to archive the file in a tar.gz format
+3. Add the archive file as an asset in the release
+
+#### macOS
+
+1. Run `./bin/build_binary -l`. Once this is run, a `dist` folder will be created with the executable in it
```

Comment:
use full-name flag if possible

Reply:
```
BIN_BUILD_CMD="pyinstaller --onefile pkg_bin/conjur"

if [[ "$1" == "-l" ]]; then
  $BIN_BUILD_CMD
  exit 0
fi
```

so it packs the cli with this script otherwise spins up a container and packs the executable in that container. So I think -l is local.
 I will add a long form